plugins/arUnogPreservicaPlugin/modules/arUnogPreservicaPlugin/actions/digitalobjectsCreateAction.class.php

```diff
@@ -0,0 +1,148 @@
+
+            throw new QubitApiUnknownException($message);
+        }
+
+        // Fail if no object with that UUID exists
+        if (empty($objectData = $client->getObjectDetails($uuid))) {
+            $message = $this->context->i18n->__('Error fetching object data.');
+
+            throw new QubitApiUnknownException($message);
+        }
+
+        $cmisId = $objectData->value->id;
+        $filename = $client->getObjectDetailsPropertyByName($objectData, 'cmis:contentStreamFileName');
+
+        // Download Preservica digital object's thumbnail to a temp file
+        $thumbTempFilePath = $client->downloadThumbnailToTempDir($cmisId, $filename);
+
+        // Create digital object with thumbnail as representation derivative
+        $do = new QubitDigitalObject();
+        $do->objectId = $io->id;
+        $do->importFromFile($thumbTempFilePath);
+        $do->save();
+
+        // Remove thumbnail temp dir and file
+        unlink($thumbTempFilePath);
+        rmdir(dirname($thumbTempFilePath));
+
+        // Store transcript, if one exists, as digital object property
+        if (!empty($fullText = $client->getFullText($uuid))) {
+            $this->storeTranscript($fullText, $do->id);
+        }
+
+        // Add property containing Preservica UUID
+        $this->addAlternativeIdentifier($do->id, arUnogPreservicaPluginConfiguration::PRESERVICA_UUID_PROPERTY_NAME, $uuid);
+
+        // Optionally add property containing Scope ID
+        if (!empty($scopeId)) {
+            $this->addAlternativeIdentifier($io->id, 'Scope ID', $scopeId);
+        }
+
+        $this->response->setStatusCode(201);
+
+        // Update information object's Elasticsearch document
+        QubitSearch::getInstance()->update($io, ['updateDescendants' => true]);
+
+        return ['id' => (int) $do->id, 'slug' => $do->slug];
+    }
+
+    private function addAlternativeIdentifier($objectId, $label, $id)
+    {
+        $property = new QubitProperty();
+        $property->objectId = $objectId;
+        $property->name = $label;
+        $property->scope = 'alternativeIdentifiers';
+
+        $property->setValue($id, ['sourceCulture' => true]);
+        $property->save();
+    }
+
+    private function storeTranscript($text, $doId)
+    {
+        // NOTE:
+        //
+        // This logic is replicated from the QubitDigitalObject model.
+        // If we add this functionality to upstream AtoM we should create
+        // a static method in the QubitDigitalObject model and use that.
+
+        // Truncate PDF text to <64KB to fit in `property.value` column
+        $text = mb_strcut($text, 0, 65535);
+
+        // Update or create 'transcript' property
+        $criteria = new Criteria();
+        $criteria->add(QubitProperty::OBJECT_ID, $doId);
+
+        $criteria->add(QubitProperty::NAME, 'transcript');
+        $criteria->add(QubitProperty::SCOPE, 'Text extracted from source PDF file\'s text layer using pdftotext');
```

Comment:
Is this scope description accurate for OCR text coming from Preservica?

Reply:
Fixed.